hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -62,6 +81,57 @@ interface HiveOperationsBase {
 
   String table();
 
+  String catalogName();
+
+  String entityType();
+
+  BaseMetastoreOperations.CommitStatus validateNewLocationAndReturnCommitStatus(
+      BaseMetadata metadata, String newMetadataLocation);
+
+  default Table loadHmsTable() throws TException, InterruptedException {
+    try {
+      return metaClients().run(client -> client.getTable(database(), table()));
+    } catch (NoSuchObjectException nte) {
+      LOG.trace("{} not found {}", entityType(), fullName(), nte);
+      return null;
+    }
+  }
+
+  void setHmsParameters(
+      BaseMetadata metadata,
```

Comment:
why does this require `BaseMetadata` as a parameter? I'm not convinced that this is the right refactoring. For views this method just passes uuid + schema.

Reply:
@nastra ,  in Case of `table` it requires to set all the table related metadata. 
```
setSnapshotStats(metadata, parameters);
    setPartitionSpec(metadata, parameters);
    setSortOrder(metadata, parameters);
```

One of the primary reason to introduce `BaseMetadata` . Please let me know what do you propose to `setHmsParameters` for `Table`.  Please take a look [here](https://github.com/apache/iceberg/pull/9852/files#diff-e502621d52f86cf0ec3187dda30ac61f6b76efb7b6276bc8d233ccb2c836fb98R187) for reference.

Reply:
I understand that for tables you have to configure more things and my comment is not about tables, but for views there's no reason to introduce this abstraction just to be able to pass the schema of the view.

Reply:
We do have [setCommonHmsParameters](https://github.com/apache/iceberg/pull/9852/files#diff-5ecfc223b311b523a12be7482b6235318fdd7535c371f899daf7e09eacddad77R123) which is being used by both table and view. Infact the same can be used directly by view. But we have to add a conditional statement.

Reply:
I guess we can explitily make this method take in everything that it needs to set, will that be better?  (instead of BaseMetadata, pass in Schema, UUID, snapshot, etc).  There is a non-trivial number of params though.